com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -92,13 +93,14 @@ protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exceptio
                 .metadata(service.getMetadata())
                 .name(service.getName())
                 .build())
-            .collect(Collectors.toMap(CloudEntity::getName, e -> e));
+            .collect(Collectors.toList());
     }
 
-    private List<String> getServicesWithoutData(List<String> servicesToDelete, Map<String, CloudServiceExtended> servicesData) {
-        return servicesToDelete.stream()
-            .filter(name -> servicesData.get(name) == null)
+    private Collection<String> getServicesWithoutData(List<String> servicesToDelete, List<CloudServiceExtended> servicesData) {
+        List<String> servicesNames = servicesData.stream()
+            .map(CloudServiceExtended::getName)
             .collect(Collectors.toList());
+        return CollectionUtils.disjunction(servicesToDelete, servicesNames);

```

Comment:
suggestion: you can make a List out of it new LinkedList<String>(collection);

Reply:
done